### ISSUE TITLE
Does `list` get all possible objects?

### ISSUE THREAD

**Issue by doivosevic** [ThouCheese/cloud-storage-rs#64]
Am I correct to assume that Object::list actually fetches all items? The `ListRequest` object accepts the `page_token` and `max_results` but looking at 
```
    pub async fn list(
        bucket: &str,
        list_request: ListRequest,
    )
```
it actually does iterate over page_tokens and fetches the whole list. I really wish this was more clear somewhere

**Comment by ThouCheese** [comment]
You are right, this is not clearly documented. The `page_token` is used as the offset starting from which all objects are fetched, and the `max_results` is currently not used. I will publish a new version of the crate with a more clear explanation and that implements `max_results` properly.

**Comment by ThouCheese** [comment]
I have made `max_results` play nicely with the repeated querying done by `Object::list` and fixed the documentation. Can you confirm that the documentation in master is clear this way?

**Comment by ThouCheese** [comment]
I have released this in version `0.8.2`.
